pkg/groups/handlers.go

```diff
@@ -499,6 +500,29 @@ func (a *API) handleAssignRoles(w http.ResponseWriter, r *http.Request) {
 
 	}
 
+	canAssign, err := a.service.CanAssignRoles(r.Context(), principal.Identifier(), roles.Roles...)
```

Comment:
I would have expected to have these checks hooked up in the converter https://github.com/canonical/identity-platform-admin-ui/blob/main/internal%2Fauthorization%2Fconverters.go#L392

main tradeoff, with the current codebase, would be that we would have to move the payload types in a centralised package

at the same time keeping this away from the handler layer and have it only in the middlewares makes it "cleaner" and keeps the handlers dumb